controller/src/beerocks/master/tasks/optimal_path_task.cpp

```diff
@@ -328,20 +328,11 @@ void optimal_path_task::work()
                     TASK_LOG(DEBUG)
                         << "requested 11K beacon measurement request from sta: " << sta_mac
                         << " on bssid: " << vap_mac;
-                    auto request = message_com::create_vs_message<
-                        beerocks_message::cACTION_CONTROL_CLIENT_BEACON_11K_REQUEST>(cmdu_tx, id);
 
-                    if (request == nullptr) {
-                        LOG(ERROR)
-                            << "Failed building ACTION_CONTROL_CLIENT_BEACON_11K_REQUEST message!";
-                        break;
+                    if (!son::son_actions::send_beacon_metrics_query_msg(
+                            cmdu_tx, database, tlvf::mac_from_string(sta_mac))) {
+                        LOG(ERROR) << "Failed sending BEACON_METRICS_QUERY_MESSAGE";
                     }
-                    request->params() = measurement_request;
-
```

Comment:
This does not make any sense. There is a state on the task that set the measurement parameters, and you replaced it with calling to a function that sends measurement request only on the bssid that the client is connected to.
This will break the optimal path task.